test.py

```diff
@@ -28,6 +28,34 @@ def test_deparam(self):
             good = base + good
             self.assertEqual(url.parse(bad).deparam(['c']).utf8(), good)
 
+    def test_deparam_case_insensitivity(self):
+        '''Parameters to remove can be supplied in any case.'''
+        examples = [
+            ('?hELLo=2', ''),
+            ('?HELLo=2', '')
+        ]
+        base = 'http://testing.com/page'
+        for bad, good in examples:
+            bad = base + bad
+            good = base + good
+            self.assertEqual(url.parse(bad).deparam(['HeLlO']).utf8(), good)
+
+    def test_filter_params(self):
+        '''We can filter parameters.'''
+        def function(name, value):
```

Comment:
Across the reaches of space and time, I can already hear brandon pointing out that url parameters technically form a key-value bag, not a map (so, a single key can appear many times.)  Maybe something to keep in mind if you go the not-`dict` route.